src/adagrams.js

```diff
@@ -1,8 +1,140 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
+class Adagrams {
+  static drawLetters() {
+    const letterValues = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
+    //make an array containing all available letters 
+    let sample = Object.entries(letterValues).map(([key , value]) => {
+      return key.repeat(value).split('');
+    });
+
+    sample = sample.flat();
+
+    const lettersAtHand = [];
+
+    for (let i = 0; i < 10; i++){
+      lettersAtHand.push(sample[Math.floor(Math.random()* sample.length)]);
+    };
+    return lettersAtHand;  
+  }
+
+  static usesAvailableLetters(input, lettersInHand){
+    const inputLetters = Array.from(input);
+
+    //make an object that hold all the letters in the lettersInHand and how many times they appear
+    const lettersAppear = {};
+
+    lettersInHand.forEach(letter => {
+      if (letter in lettersAppear) {
+        lettersAppear[letter] += 1;
+      } else {
+        lettersAppear[letter] = 1;
+      }
+    });
+
+    //make another loop that will go through the object and decrement if it exists 
+    for (let i in inputLetters ){
```

Comment:
Good use of a for...in loop so that you're able to return out of it when you encounter a letter that's not in hand.